### PR TITLE
 Boxplot Options

### PR DESCRIPTION
**Pull request by Fidel365**
Fixes(partially) #7556 
Procurement>Describe>Numeric>Boxplot
@N-thony , this PR is ready for review

### CODE REVIEW

**Comment by Fidel365** [comment]
@N-thony , the recent commit fixes minor overlapping issues

**Comment by lloyddewit** [comment]
@rdstern please could you test? thanks

**Comment by rdstern** [comment]
@Fidel365 I note this (and some other of your pull requests seem to overlap with those by @EstherNjeriLiberatta .  For this one - boxplots - how do your changes relate to her pull request in #7570 ?

**Comment by Fidel365** [comment]
@rdstern , we are working on the same dialogues but from different menus, we are trying to update all dialogues to geom options.e.g The boxplot commit follows the path Climatic>Describe>Boxplot

**Comment by rdstern** [comment]
@Fidel365 thank you for the clarification.  It was just that at the top of your pull request, you mention the `Procurement menu`.  The boxplot dialogue there is just the same as the main one.  That's unlike the `Climatic > Check Data > Boxplot `that you have now mentioned and that is using a different dialogue - so great that you are adapting that one.  I'll check it now.

**Comment by N-thony** [comment]
@EstherNjeriLiberatta can you test/review this? Thanks.

File: instat/dlgBoxPlot.vb [hunk 1]
```diff
@@ -317,81 +317,30 @@ Public Class dlgBoxplot
...
         If rdoBoxplotTufte.Checked Then
             If ucrChkTufte.Checked Then
-                cmdBoxPlotOptions.Text = "Tufte Box Options"
+                'cmdBoxPlotOptions.Text = "Tufte Box Options"
                 ucrSaveBoxplot.SetPrefix("tufte_boxplot")
                 clsCurrGeomFunc = clsTufteBoxplotFunc
                 clsStatSummary.AddParameter("size", 0.7, iPosition:=3)
                 ucrSecondFactorReceiver.ChangeParameterName("colour")
             Else
-                cmdBoxPlotOptions.Text = "Box Options"
```
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:382]: This too

File: instat/dlgBoxPlot.vb [hunk 2]
```diff
@@ -317,81 +317,30 @@ Public Class dlgBoxplot
...
+                'cmdBoxPlotOptions.Text = "Box Options"
                 ucrSaveBoxplot.SetPrefix("box_plot")
                 ucrSecondFactorReceiver.ChangeParameterName("fill")
                 clsCurrGeomFunc = clsBoxplotFunc
             End If
 
         ElseIf rdoJitter.Checked Then
-            cmdBoxPlotOptions.Text = "Jitter Options"
```
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:389]: And this

File: instat/dlgBoxPlot.vb [hunk 3]
```diff
@@ -317,81 +317,30 @@ Public Class dlgBoxplot
...
+            'cmdBoxPlotOptions.Text = "Jitter Options"
             ucrSaveBoxplot.SetPrefix("jitter")
             ucrSecondFactorReceiver.ChangeParameterName("colour")
             clsCurrGeomFunc = clsJitterplotFunc
         Else
-            cmdBoxPlotOptions.Text = "Violin Options"
```
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:394]: This one too

File: instat/dlgBoxPlot.vb [hunk 4]
```diff
@@ -405,10 +354,10 @@ Public Class dlgBoxplot
 
     Private Sub TempOptionsDisabledInMultipleVariablesCase()
         If ucrVariablesAsFactorForBoxplot.bSingleVariable Then
-            cmdBoxPlotOptions.Enabled = True
```
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:408]: Here as well

File: instat/dlgBoxPlot.vb [hunk 5]
```diff
@@ -405,10 +354,10 @@ Public Class dlgBoxplot
 
     Private Sub TempOptionsDisabledInMultipleVariablesCase()
         If ucrVariablesAsFactorForBoxplot.bSingleVariable Then
-            cmdBoxPlotOptions.Enabled = True
+            'cmdBoxPlotOptions.Enabled = True
             cmdOptions.Enabled = True
         Else
-            cmdBoxPlotOptions.Enabled = False
```
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:411]: And here

File: instat/dlgBoxPlot.vb [hunk 6]
```diff
@@ -433,6 +382,56 @@ Public Class dlgBoxplot
...
+    Private Sub toolStripMenuItemBoxOptions_Click(sender As Object, e As EventArgs) Handles toolStripMenuItemBoxOptions.Click
+        'SetupLayer sends the components storing the plot info (clsRgeom_boxplotFunction, clsRggplotFunction, ...) of dlgBoxPlot through to sdgLayerOptions where these will be edited.
+        sdgLayerOptions.SetupLayer(clsNewGgPlot:=clsRggplotFunction, clsNewGeomFunc:=clsCurrGeomFunc, clsNewGlobalAesFunc:=clsRaesFunction, clsNewLocalAes:=clsLocalRaesFunction, bFixGeom:=True, ucrNewBaseSelector:=ucrSelectorBoxPlot, bApplyAesGlobally:=True, bReset:=bResetBoxLayerSubdialog)
+        sdgLayerOptions.ShowDialog()
+        bResetBoxLayerSubdialog = False
+        'Coming from the sdgLayerOptions, clsRgeom_boxplot and others has been modified. One then needs to display these modifications on the dlgBoxPlot.
+        If clsCurrGeomFunc.GetParameter("varwidth") IsNot Nothing Then
+            If clsCurrGeomFunc.GetParameter("varwidth").strArgumentValue = "TRUE" Then
+                ' chkVarwidth.Checked = True
+                'Observe that changing the check of the chkVarwidth here doesn't trigger the checkchanged event.
```
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:408]: These lines

File: instat/dlgBoxPlot.vb [hunk 7]
```diff
@@ -433,6 +382,56 @@ Public Class dlgBoxplot
...
+            End If
+        Else
+            'chkVarwidth.Checked = False
```
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:411]: This too

**Review by EstherNjeriLiberatta — CHANGES_REQUESTED** [review]

File: instat/dlgBoxPlot.vb [hunk 8]
```diff
@@ -317,81 +317,30 @@ Public Class dlgBoxplot
...
-        'ucrVariablesAsFactorForBoxplot.SetReceiverStatus()
-    End Sub
-
     Private Sub SetGeomPrefixFillColourAes()
         'Sets geom function, fill and colour aesthetics, ucrsave prefix and stat summary parameters
         clsStatSummary.AddParameter("size", 1.5, iPosition:=3)
         If rdoBoxplotTufte.Checked Then
             If ucrChkTufte.Checked Then
-                cmdBoxPlotOptions.Text = "Tufte Box Options"
+                'cmdBoxPlotOptions.Text = "Tufte Box Options"
```
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:325]: You could remove this because it's not needed.

File: instat/dlgBoxPlot.vb [hunk 9]
```diff
@@ -317,81 +317,30 @@ Public Class dlgBoxplot
...
                 ucrSaveBoxplot.SetPrefix("tufte_boxplot")
                 clsCurrGeomFunc = clsTufteBoxplotFunc
                 clsStatSummary.AddParameter("size", 0.7, iPosition:=3)
                 ucrSecondFactorReceiver.ChangeParameterName("colour")
             Else
-                cmdBoxPlotOptions.Text = "Box Options"
+                'cmdBoxPlotOptions.Text = "Box Options"
```
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:331]: This too

File: instat/dlgBoxPlot.vb [hunk 10]
```diff
@@ -317,81 +317,30 @@ Public Class dlgBoxplot
...
                 ucrSaveBoxplot.SetPrefix("box_plot")
                 ucrSecondFactorReceiver.ChangeParameterName("fill")
                 clsCurrGeomFunc = clsBoxplotFunc
             End If
 
         ElseIf rdoJitter.Checked Then
-            cmdBoxPlotOptions.Text = "Jitter Options"
+            'cmdBoxPlotOptions.Text = "Jitter Options"
```
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:338]: This one too

File: instat/dlgBoxPlot.vb [hunk 11]
```diff
@@ -317,81 +317,30 @@ Public Class dlgBoxplot
...
             ucrSaveBoxplot.SetPrefix("jitter")
             ucrSecondFactorReceiver.ChangeParameterName("colour")
             clsCurrGeomFunc = clsJitterplotFunc
         Else
-            cmdBoxPlotOptions.Text = "Violin Options"
+            'cmdBoxPlotOptions.Text = "Violin Options"
```
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:343]: This one as well

File: instat/dlgBoxPlot.vb [hunk 12]
```diff
@@ -405,10 +354,10 @@ Public Class dlgBoxplot
 
     Private Sub TempOptionsDisabledInMultipleVariablesCase()
         If ucrVariablesAsFactorForBoxplot.bSingleVariable Then
-            cmdBoxPlotOptions.Enabled = True
+            'cmdBoxPlotOptions.Enabled = True
```
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:357]: Here

File: instat/dlgBoxPlot.vb [hunk 13]
```diff
@@ -405,10 +354,10 @@ Public Class dlgBoxplot
 
     Private Sub TempOptionsDisabledInMultipleVariablesCase()
         If ucrVariablesAsFactorForBoxplot.bSingleVariable Then
-            cmdBoxPlotOptions.Enabled = True
+            'cmdBoxPlotOptions.Enabled = True
             cmdOptions.Enabled = True
         Else
-            cmdBoxPlotOptions.Enabled = False
+            'cmdBoxPlotOptions.Enabled = False
```
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:360]: And here

File: instat/dlgBoxPlot.vb (hunk 6 above)
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:408]: These lines?

File: instat/dlgBoxPlot.vb (hunk 7 above)
Comment (EstherNjeriLiberatta) [inline instat/dlgBoxPlot.vb:411]: This too

**Review by EstherNjeriLiberatta — CHANGES_REQUESTED** [review]

**Comment by N-thony** [comment]
@derekagorhom can you take this over?

**Comment by derekagorhom** [comment]
> @derekagorhom can you take it over?

@Fidel365 still working on it, i will help him with the corrections

**Comment by Fidel365** [comment]
@EstherNjeriLiberatta , this PR is ready for a rereview

**Comment by lloyddewit** [comment]
> @EstherNjeriLiberatta , this PR is ready for a rereview

@EstherNjeriLiberatta I understand that you're currently working on other projects. I checked the new commit and all your comments above are resolved.
To avoid potential delays, I'm happy to peer review this PR.

**Comment by Fidel365** [comment]
@lloyddewit , it's ready for testing

**Comment by lloyddewit** [comment]
@rdstern Please can you test? thanks

**Review by rdstern — CHANGES_REQUESTED** [review]
@N-thony it would be good to re-allocate this work now.
1) Please note that the dialogue has now got out-of line - compare the position of the receivers for the Points and the Group option in 0.7.7 and the new one.  I give both below:
image
And in the new one:
image
 The Add Points and Groups to Connect should move back to their original positions.
b) Change Box to Boxplot
c) Add the other options (Tufte, Jitter, Violin) into the pull-down.  Note that the work has been, because when I move to Jitter or Tufte or Violin and press on Box it goes to the right place.
d) I don't mind if they all remain available all the time.  But if you are disabling those that don't apply, then note that you should enable Jitter, (as well as Boxplot) when you Add Points.  
e) Also add Summary when you join the means/medians and go to stat_summary.?

**Comment by rdstern** [comment]
@N-thony after the above I am closing this one now, and continuing with #7570 .
Please reopen if I have got that wrong.

**Pull request closed without merging**